complex/ft_main.c

```diff
@@ -335,6 +335,9 @@ static int ft_fw_process_list(struct fi_info *hints, struct fi_info *info)
 			return ret;
 
 		ft_fw_update_info(&test_info, fabric_info, subindex);
+		if (strcmp(hints->fabric_attr->prov_name, (&test_info)->prov_name))
+			continue;
+
```

Comment:
This can be moved to the very the beginning of the loop where hints->fabric_attr->prov_name can be compared with fabric_info->fabric_attr->prov_name. This would avoid extra processing afterwards.

Also ft_show_test_info should be moved below ft_fw_update_info so that it prints the updated test_info.